src/strategy/texture2D.ts

```diff
@@ -42,47 +46,45 @@ export const applyTexture2DStrategy: ApplyStrategy<Texture2DStrategy> = (
           }
 
           const name = generate(path.node.args[0]);
-          seen[name] = (seen[name] || 0) + 1;
-          texture2Dcalls.push([
-            name,
-            path.parent as AstNode,
-            path.key,
-            // Remove the first argument and comma
-            (path.node.args as AstNode[]).slice(2),
-          ]);
+
+          if (!(name in references)) {
+            references[name] = [];
+            texture2Dcalls.push(name);
+          }
+
+          references[name].push({
+            parent: path.parent as AstNode,
+            key: path.key,
+            args: (path.node.args as AstNode[]).slice(2),
+          });
         }
       },
     },
   };
   visit(ast, visitors);
-  const names = new Set(
-    Object.entries(seen).reduce<string[]>(
-      (arr, [name, count]) => [...arr, ...(count > 1 ? [name] : [])],
-      [],
-    ),
-  );
-  const inputs = texture2Dcalls.map<ComputedInput>(
-    ([name, parent, key, texture2dArgs], index) => {
-      // Suffix input name if it's used more than once
-      const iName = names.has(name) ? `${name}_${index}` : name;
-      return [
-        nodeInput(
-          iName,
-          `filler_${iName}`,
-          'filler',
-          'vector4', // Data type for what plugs into this filler
-          ['code', 'data'],
-          false,
-        ),
-        (fillerAst) => {
+
+  const inputs = texture2Dcalls.map<ComputedInput>((name) => {
+    return [
+      nodeInput(
+        name,
+        `filler_${name}`,
+        'filler',
+        'vector4', // Data type for what plugs into this filler
+        ['code', 'data'],
+        false,
+      ),
+      (filler) => {
+        references[name].forEach(({ parent, key, args }) => {
+          // Backfilling into the filler! Similar to parsers.ts filler
+          const f = filler(...args.map(generate));
```

Comment:
This is the first most important part of backfilling. This takes the arguments from this call:

```glsl
vec4 color = texture2D(image1, vUv * 2.0);
```
and preserves `vUv * 2.0`, so that when this input is filled, it calls the filler (now a **function**) with these arguments. If the filler is the main function of another shader, this line becomes

```glsl
vec4 color = shader_main(vUv * 2.0);
```

See the other "important" github comment for how this variable is then injected into the main function of the shader for use.